sklearn/gaussian_process/kernels.py

```diff
@@ -393,6 +398,27 @@ def is_stationary(self):
         return True
 
 
+class VectorOnlyKernelMixin:
+    """Mixin for kernels which operate on fixed-length feature vectors
+
+    .. versionadded:: TBD
+    """
+    def requires_vector_input(self):
+        """whether the kernel only works on fixed-length feature vectors."""
+        return True
+
+
+class StructureOrGenericKernelMixin:
```

Comment:
Would this usually be called a "Structured Kernel" rather than a "Structure Kernel"?

Can we just call this "GenericKernelMixin"?

Reply:
TL;DR: sure we can call it a 'GenericKernelMixin'. Will update accordingly.